### ISSUE TITLE
Parallel

### ISSUE THREAD

**Issue by JorgeCastilloPrz** [arrow-kt/arrow#1091]
`Parallel` is a Type class resolving parallel computation using Monads. Since they have the power of both sequential and parallel computation, they require an instance of Monad and an instance of Applicative.

`NonEmptyParallel` is also required on this one.

* [Here is the cats implementation](https://github.com/typelevel/cats/blob/9807343a553710b4df33504651803fcc8fa20d99/core/src/main/scala/cats/Parallel.scala#L57) for reference.
* [Here you have cats docs for better context](https://github.com/typelevel/cats/blob/master/docs/src/main/tut/typeclasses/parallel.md)

**Important:** After #1043, we left `IOParallel` TODOs on some async calls without wiring the recently added `IOConnection` for cancellation and keeping the old cancellation style. That should be fixed whenever `Parallel` is implemented.

**UPDATE (Also discussed [here](https://github.com/typelevel/cats-effect/issues/430#issuecomment-445214878))**.

`parMapN` should be available on `Concurrent` instead of on a separate typeclass `Parallel`. Since `Concurrent` defines `start`, `racePair` & `raceN` already, it should also define `parMapN`. This also reduces awkwardness currently involved in using `Parallel` in cats.

Sidenote: AFAIK `Parallel` is in cats because it existed from before `cats-effects` came to live.

Therefore, it'll be done inside Concurrent but keep being a separate issue as is.

**Comment by nomisRev** [comment]
Also discussed [here](https://github.com/typelevel/cats-effect/issues/430#issuecomment-445214878).

`parMapN` should be available on `Concurrent` instead of on a separate typeclass `Parallel`. Since `Concurrent` defines `start`, `racePair` & `raceN` already, it should also define `parMapN`. This also reduces awkwardness currently involved in using `Parallel` in cats.

Sidenote: AFAIK `Parallel` is in cats because it existed from before `cats-effects` came to live.

**Comment by JorgeCastilloPrz** [comment]
Looks like parMapN is already there since some time ago, but we're still lacking other things like parSequence, parTraverse, and maybe paraTupled. All of them working over F.

Could you take a careful look to what's still missing and see if we can add it @nomisRev?

Thanks in advance!

**Comment by JorgeCastilloPrz** [comment]
Closing this. After discussing with @nomisRev looks like we got all the combinators we strictly need ready, and any ones that we could miss in the future could be easily derived from those. So we'd open this issue again or a new one in the future if we found the need for further combinators.
